client/www/components/docs/TabbedSingle.jsx

```diff
@@ -0,0 +1,89 @@
+import { useState, useContext } from 'react';
+import { ClipboardDocumentIcon } from '@heroicons/react/24/outline';
+import { SelectedAppContext } from '@/lib/SelectedAppContext';
+import useLocalStorage from '@/lib/hooks/useLocalStorage';
+
+/**
+ * TabbedSingle - A component for displaying single-line content with tab selection.
+ *
+ * Useful for things like showiing one-liners for different platforms /
+ * framworks / etc. (e.g. Next.js, Expo, Vanilla TS, etc.)
+ *
+ * @param {Object} tabs - Object with tab keys containing { label, content }
+ * @param {string} defaultTab - Optional default selected tab key
+ * @param {string} storageKey - Optional unique key for localStorage persistence
+ */
+export function TabbedSingle({ tabs, defaultTab, storageKey }) {
+  // Defensive check for tabs
+  if (!tabs || Object.keys(tabs).length === 0) {
+    return null;
+  }
+
+  const defaultValue = defaultTab || Object.keys(tabs)[0];
+
+  // Use localStorage if storageKey is provided, otherwise use regular state
+  const [selectedTab, setSelectedTab] = storageKey
+    ? useLocalStorage(`tabbed-single-${storageKey}`, defaultValue)
```

Comment:
This goes against the rules of hooks -- you can't conditionally render hooks. If `storageKey` ever changed this would throw an error. I would just settle on localStorage